test/jdk/tools/jpackage/macosx/SigningRuntimeImagePackageTest.java

```diff
@@ -0,0 +1,263 @@
+ * note:
+ * "jpackage.openjdk.java.net" can be over-ridden by systerm property
+ * "jpackage.mac.signing.key.user.name", and
+ * "jpackagerTest" can be over-ridden by system property
+ * "jpackage.mac.signing.keychain"
+ */
+
+/*
+ * @test
+ * @summary jpackage with --type pkg,dmg --runtime-image --mac-sign
+ * @library /test/jdk/tools/jpackage/helpers
+ * @library base
+ * @key jpackagePlatformPackage
+ * @build SigningBase
+ * @build jdk.jpackage.test.*
+ * @build SigningRuntimeImagePackageTest
+ * @requires (jpackage.test.MacSignTests == "run")
+ * @run main/othervm/timeout=720 -Xmx512m jdk.jpackage.test.Main
+ *  --jpt-run=SigningRuntimeImagePackageTest
+ *  --jpt-before-run=SigningBase.verifySignTestEnvReady
+ */
+public class SigningRuntimeImagePackageTest {
+
+    private static void verifyPKG(JPackageCommand cmd) {
+        Path outputBundle = cmd.outputBundle();
+        SigningBase.verifyPkgutil(outputBundle, isPKGSigned(cmd), getCertIndex(cmd));
+        if (isPKGSigned(cmd)) {
+            SigningBase.verifySpctl(outputBundle, "install", getCertIndex(cmd));
+        }
+    }
+
+    private static void verifyDMG(JPackageCommand cmd) {
+        Path outputBundle = cmd.outputBundle();
+        SigningBase.verifyDMG(outputBundle);
+    }
+
+    private static void verifyRuntimeImageInDMG(JPackageCommand cmd,
+                                                boolean isRuntimeImageSigned,
+                                                int JDKBundleCertIndex) {
+        MacHelper.withExplodedDmg(cmd, dmgImage -> {
+            Path launcherPath = ApplicationLayout.platformAppImage()
+                    .resolveAt(dmgImage).launchersDirectory().resolve("libjli.dylib");
+            // We will be called with all folders in DMG since JDK-8263155, but
+            // we only need to verify JDK bundle.
+            if (dmgImage.endsWith(cmd.name() + ".jdk")) {
+                SigningBase.verifyCodesign(launcherPath, isRuntimeImageSigned,
+                        JDKBundleCertIndex);
+                SigningBase.verifyCodesign(dmgImage, isRuntimeImageSigned,
+                        JDKBundleCertIndex);
+                if (isRuntimeImageSigned) {
+                    SigningBase.verifySpctl(dmgImage, "exec", JDKBundleCertIndex);
+                }
+            }
+        });
+    }
+
+    private static boolean isPKGSigned(JPackageCommand cmd) {
+        return cmd.hasArgument("--mac-signing-key-user-name") ||
+               cmd.hasArgument("--mac-installer-sign-identity");
+    }
+
+    private static int getCertIndex(JPackageCommand cmd) {
+        if (cmd.hasArgument("--mac-signing-key-user-name")) {
+            String devName = cmd.getArgumentValue("--mac-signing-key-user-name");
+            return SigningBase.getDevNameIndex(devName);
+        } else {
+            return SigningBase.CertIndex.INVALID_INDEX.value();
+        }
+    }
+
+    private static Path getRuntimeImagePath(boolean useJDKBundle,
```

Comment:
The name of the function is misleading. It doesn't get the runtime image. It creates it.